pkg/querier/querier.go

```diff
@@ -221,12 +204,27 @@ func (qm MergeQuerier) RemoteReadHandler(w http.ResponseWriter, r *http.Request)
 				return
 			}
 
-			matrix, err := qm.Query(ctx, from, to, matchers...)
+			iterators, err := qm.QueryRange(ctx, from, to, matchers...)
 			if err != nil {
 				errors <- err
 				return
 			}
 
+			in := metric.Interval{
+				OldestInclusive: from,
+				NewestInclusive: to,
+			}
+
+			// Convert iterators to matrix
+			matrix := make(model.Matrix, 0, len(iterators))
+			for _, it := range iterators {
+				ss := &model.SampleStream{
+					Metric: it.Metric().Metric,
+					Values: it.RangeValues(in),
```

Comment:
It looks like we only pass these to [`ToQueryResponse`](https://github.com/weaveworks/cortex/blob/416-lazy-series-when-no-metric-name/pkg/util/compat.go#L77) but we don't check for `nil` there, so we could potentially "[panic](https://play.golang.org/p/SK_rgQtRBb)".